AppCenterDistribute/AppCenterDistributeTests/MSDistributeCheckForUpdateTests.m

```diff
@@ -4,111 +4,278 @@
+  [self.keychainUtilMock stopMocking];
+  [self.bundleMock stopMocking];
+  [self.parserMock stopMocking];
+  self.sut = nil;
+}
+
+- (void)testBypassCheckForUpdateWhenDistributeIsDisabled {
+
+  // If
+  id distributeMock = OCMPartialMock(self.sut);
+  OCMStub([distributeMock canBeUsed]).andReturn(YES);
+  OCMStub([distributeMock isEnabled]).andReturn(NO);
+  [self.settingsMock removeObjectForKey:kMSUpdateTokenRequestIdKey];
+  OCMReject([distributeMock startUpdateOnStart:OCMOCK_ANY]);
+
+  // When
+  [MSDistribute checkForUpdate];
+
+  // Then
+  OCMVerifyAll(distributeMock);
+
+  // Clear
+  [distributeMock stopMocking];
+}
+
+- (void)testBypassCheckForUpdateWhenDistributeCanNotBeUsed {
+
+  // If
+  id distributeMock = OCMPartialMock(self.sut);
+  OCMStub([distributeMock canBeUsed]).andReturn(NO);
+  OCMStub([distributeMock isEnabled]).andReturn(YES);
+  [self.settingsMock removeObjectForKey:kMSUpdateTokenRequestIdKey];
+  OCMReject([distributeMock startUpdateOnStart:OCMOCK_ANY]);
+
+  // When
+  [MSDistribute checkForUpdate];
+
+  // Then
+  OCMVerifyAll(distributeMock);
+
+  // Clear
+  [distributeMock stopMocking];
+}
+
+- (void)testBypassCheckForUpdateIfUpdateRequestIdTokenExists {
+
+  // If
+  id distributeMock = OCMPartialMock(self.sut);
+  OCMStub([distributeMock canBeUsed]).andReturn(YES);
+  OCMStub([distributeMock isEnabled]).andReturn(YES);
+  [self.settingsMock setObject:@"testToken" forKey:kMSUpdateTokenRequestIdKey];
+  OCMReject([distributeMock startUpdateOnStart:OCMOCK_ANY]);
+
+  // When
+  [MSDistribute checkForUpdate];
+
+  // Then
+  OCMVerifyAll(distributeMock);
+
+  // Clear
+  [distributeMock stopMocking];
 }
 
 - (void)testDisableAutomaticCheckForUpdateBeforeStart {
 
   // If
-  MSDistribute *distribute = [MSDistribute sharedInstance];
-  distribute.automaticCheckForUpdateDisabled = NO;
+  self.sut.automaticCheckForUpdateDisabled = NO;
 
   // When
   [MSDistribute disableAutomaticCheckForUpdate];
 
   // Then
-  XCTAssertTrue(distribute.automaticCheckForUpdateDisabled);
+  XCTAssertTrue(self.sut.automaticCheckForUpdateDisabled);
 }
 
 - (void)testAutomaticCheckForUpdateDisabledDoesNotChangeAfterStart {
 
   // If
-  MSDistribute *distribute = [MSDistribute sharedInstance];
-  [distribute startWithChannelGroup:OCMProtocolMock(@protocol(MSChannelGroupProtocol))
-                          appSecret:kMSTestAppSecret
-            transmissionTargetToken:nil
-                    fromApplication:YES];
+  [self.sut startWithChannelGroup:OCMProtocolMock(@protocol(MSChannelGroupProtocol))
+                        appSecret:kMSTestAppSecret
+          transmissionTargetToken:nil
+                  fromApplication:YES];
+
+  // When
+  [MSDistribute disableAutomaticCheckForUpdate];
+
+  // Then
+  XCTAssertFalse(self.sut.automaticCheckForUpdateDisabled);
+}
+
+- (void)testCheckForUpdateOpenBrowserEvenThoughAutomaticUpdateIsDisabled {
+
+  // If
+  id utilityMock = OCMClassMock([MSUtility class]);
```

Comment:
Utility mock should be shared.  It's used by 4+ tests now and seems to be mostly set up the same way.